### DISCUSSION TITLE
Array of string as options?

### DISCUSSION THREAD

**Discussion by kaiserfl** [JedWatson/react-select#4956] [Q&A]
In version 4.x you could use an string array as options, if you used the identity functions for getOptionValue.
In version 5.x this leads to an TypeError: "Cannot use 'in' operator to search for 'options' in X"

Question: Is this a Bug or am I using it wrong?

**Comment by Rall3n** [comment]
Hello @kaiserfl,

`react-select` has not supported string options directly since the release of `v2.0.0`. Partial support may have existed with how the component functions internally, but never intended.

The error message came with the change from Flow-typed JavaScript to TypeScript, where some internal changes had to be made to allow for correctly typed code.

String option support (using the accessor props) may be possible, but needs some major internal changes to work correctly in all scenarios.

-----

In the meantime I suggest you either use object options, or a wrapper/hook which turns your simple options into objects:

```js
const SelectWithSimpleValues = ({ options, value, onChange }) => {
    const selectOptions = useMemo(() => options.map(o => ({ val: o })), [options]);
    
    const selectValue = useMemo(() => (value ? { val: value } : null), [value]);
    
    const onSelectChange = useCallback((value) => {
        onChange(value.val);
    }, [onChange]);

    return <Select
        options={selectOptions}
        value={selectValue}
        onChange={onSelectChange}
        getOptionValue={o => o.val}
        getOptionLabel={o => o.val}
    />;
}
```
